Use collections.Counter instead of a dict - Closes #7868

tanvimehta opened the pull request:
Use `collections.Counter` instead of a `dict` in `terminal.py` Closes #7868

<!--
Thanks for submitting a PR, your contribution is really appreciated!

Here is a quick checklist that should be present in PRs.

- [ ] Include documentation when adding new features.
- [ ] Include new tests or update existing tests when applicable.
- [X] Allow maintainers to push and squash when merging my commits. Please uncheck this if you prefer to squash the commits yourself.

If this change fixes an issue, please:

- [ ] Add text like ``closes #XYZW`` to the PR description and/or commits (where ``XYZW`` is the issue number). See the [github docs](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more information.

Unless your change is trivial or a small documentation fix (e.g., a typo or reword of a small section) please:

- [ ] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.

  Write sentences in the **past or present tense**, examples:

  * *Improved verbose diff output with sequences.*
  * *Terminal summary statistics now use multiple colors.*

  Also make sure to end the sentence with a `.`.

- [ ] Add yourself to `AUTHORS` in alphabetical order.
-->

tanvimehta commented:
@Zac-HD Would appreciate it if you could help review this. Thanks!

Zac-HD reviewed (approved):
This looks great to me - thanks @tanvimehta, and congrats on your first Pytest PR :tada: 

I'd also encourage you to add your name to the list of contributors in https://github.com/pytest-dev/pytest/blob/master/AUTHORS, and edit your PR description to [use the magic keywords](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords) to automatically close the linked issue (e.g. `Closes #7868` instead of `closes issue #7868` - putting "issue" in the middle disables it).

Then we'll be ready to merge :star_struck:

Zac-HD commented:
Apparently keywords only work in the body, not the title - you should see a little line under the keyword (and tooltip) when it's working, including in the preview.

(FYI this is not at all a blocker, I'd merge it anyway, but the point here is to level you up as a OSS contributor :sparkling_heart:)

tanvimehta commented:
Thanks a lot @Zac-HD. I added the magic word to the body(confirmed that it works as expected) and added my name to the AUTHORS file. I assumed this change would be considered trivial so I wasn't sure about adding my name to the authors' list.

Zac-HD commented:
> I assumed this change would be considered trivial so I wasn't sure about adding my name to the authors' list.

Fair enough!  From one perspective it *is* trivial, in that it probably took me longer to write it up and review your PR than it would have to just write the patch myself.  But IMO that's the *wrong* perspective - we did it this way because mentoring and community matter for us (pytest and the wider Python OSS community), and you absolutely are a contributor.  So don't be shy about adding yourself to the list on other projects - the length of that list is a point of pride for us :grin: 

-------

One bit of advice for next time: I just noticed that you made the PR from your `master` branch - it's better to [use a PR-specific branch](https://guides.github.com/introduction/flow/), which lets you work on more than one PR or feature at a time.  Later on you can also do fancy tricks like rebasing on the latest upstream version to fix any conflicts.

And finally, :sparkles:CONGRATULATIONS:sparkles: on the PR :tada:
